ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -287,10 +287,10 @@ static VirtualizationRootHandle FindUnusedIndexOrGrow_Locked()
     if (RootHandle_None == rootIndex)
     {
         // No space, resize array
-        uint16_t newLength = MIN(s_maxVirtualizationRoots * 2u, INT16_MAX + 1u);
-        if (newLength <= s_maxVirtualizationRoots)
```

Comment:
This is checking if the `newLength` is actually the same as the existing length, in which case there is no need to reallocate memory. Nothing to do with overflow.

Reply:
Walk me through how that would happen? I must be missing something.

Reply:
Oh, wait, I see. We're mixing signed and unsigned types. That begs the question -- why are we mixing signed and unsigned types?

Reply:
The negative values for root handles are the various special `RootHandle_None`, `RootHandle_Indeterminate`, etc. values. We could move those to the top end of the `uint16_t` range instead, but I suspect that would make things worse, not better. So I guess the solution would be to limit `s_maxVirtualizationRoots` to `INT16_MAX` instead of `INT16_MAX + 1` and use signed types throughout.